### DISCUSSION TITLE
v1.1.5-2-beta

### DISCUSSION THREAD

**Discussion by Aleks130699** [Aleks130699/ha-fpp#12] [General]
**Full Changelog**: https://github.com/Aleks130699/ha-fpp/compare/v1.1.5-1...v1.1.5-2-beta

<hr /><em>This discussion was created from the release <a href='https://github.com/Aleks130699/ha-fpp/releases/tag/v1.1.5-2-beta'>v1.1.5-2-beta</a>.</em>

**Comment by zackbcom** [comment]
I have some pending changes to the test branch. Might be out for a few days. 

Todo list:
- create python FPP_client module. Going to have it be in the same code base for now. 
- updates to config, use URL instead of host/port to allow greater customization. 
  - allow definition for schema, host,port, additional route (if proxies)
- updates to defaults, username to admin, default timeout, 
- change from inline auth to BasicAuth. 
- use cv.string, cv.url for validations of configs. 
-

**Reply by zackbcom** [reply]
I've created an MR that you can takes pieces from, just had my child today, will be on a big break. 
I had seeked references from the core/homeassistant/components for some of my changes.
Most notably, Radarr.

We should probably build the fpp_client so that we can make the calls.
Maybe if we get this looking good we can get it added into HA

**Reply by zackbcom** [reply]
Would be nice if we could get this setup well, we could add some additional devices/platforms.
Switches, Voltages, Calendars, Media Browser, 

Wonder how we migth be able to leverage MQTT

**Comment by zackbcom** [comment]
I started work on [fppclient](https://github.com/zackbcom/fppclient)

Still a wip, it's deployed to pypi. need to integrate
